test/autorest/generated/lrogroup/pollers.go

```diff
@@ -161,6 +167,7 @@ func (p *productArrayPoller) pollUntilDone(ctx context.Context, frequency time.D
 			return nil, err
 		}
 		if p.Done() {
+			p.resp, _ = p.handleResponse(p.pt.latestResponse())
```

Comment:
Presumably it's ok to ignore the error here because it will get returned from the call to `FinalResponse()` yes?

Reply:
That was the original intent, but taking a closer look at the unmarshalling section of handleResponse, I think this error could be hidden away. I'll add the immediate error check and return here as well.

Reply:
Done